content-moderation/04-text-moderation/02-content-moderation-text-toxic-classification.ipynb

```diff
@@ -0,0 +1,474 @@
+   "metadata": {},
+   "source": [
+    "## Toxicity Classficiation\n",
+    "\n",
+    "- [Step 1: Setup notebook](#step1)\n",
+    "- [Step 2: Prepare custom classification training dataset](#step2)\n",
+    "- [Step 3: Create Amazon Comprehend Classification training job](#step3)\n",
+    "- [Step 4: Create Amazon Comprehend real time endpoint](#step4)\n",
+    "- [Step 5: Classify Documents using the real-time endpoint](#step5)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Step 1: Setup Notebook <a id=\"step1\"></a>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# First, let's get the latest installations of our dependencies\n",
+    "%pip install --upgrade pip\n",
+    "%pip install botocore --upgrade\n",
+    "%pip install boto3 --upgrade\n",
+    "%pip install -U botocore"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can skip the below cell if you are using SageMaker Studio Data Science kernel or they are already installed in your environment."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# install pandas and numpy if you are using a local IDE and they are not installed in your env\n",
+    "%pip install pandas\n",
+    "%pip install numpy"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import boto3\n",
+    "import botocore\n",
+    "import sagemaker as sm\n",
+    "import os\n",
+    "import io\n",
+    "import datetime\n",
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "\n",
+    "# variables\n",
+    "data_bucket = sm.Session().default_bucket()\n",
+    "region = boto3.session.Session().region_name\n",
+    "\n",
+    "os.environ[\"BUCKET\"] = data_bucket\n",
+    "os.environ[\"REGION\"] = region\n",
+    "role = sm.get_execution_role()\n",
+    "\n",
+    "print(f\"SageMaker role is: {role}\\nDefault SageMaker Bucket: s3://{data_bucket}\")\n",
+    "\n",
+    "s3=boto3.client('s3')\n",
+    "comprehend=boto3.client('comprehend', region_name=region)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Step 2: Prepare custom classification training dataset <a id=\"step2\"></a>\n",
+    "Unzip the sample data **toxicity.zip** and decompress files to a local folder"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!unzip ../datasets/toxicity.zip -d toxicity_dataset"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This CSV file contains 500 toxic and 500 non-toxic comments from a variety of popular social media platforms. Click on toxicity_en.csv to see a spreadsheet of 1000 English examples.\n",
```

Comment:
This is MIT licensed right? Should we have a link here to link the source in Github or website?

Reply:
The license file is in the dataset package. I rather not refer to the GitHub repo as competitors own it...